### DISCUSSION TITLE
Behaviour of resources on platform based on start and end date

### DISCUSSION THREAD

**Discussion by Sanjana-dm** [Sunbird-Ed/Community#377] [Enhancement request]
Dear all, 

This is the document on the behaviour of resources on the platform based on their end and start date.
Examples and use cases have been taken based on projects, observations, courses, programs and surveys. 
https://docs.google.com/document/d/1SU3EDHi_T2QZ2sodbH992B5db5XHgzbR-KYpvxFsDQs/edit?usp=sharing

[@Prateek-slokam](https://github.com/Prateek-slokam) [@aks30](https://github.com/aks30) [@kiranharidas187](https://github.com/kiranharidas187) [@vijiurs](https://github.com/vijiurs) [@Sanjana-dm](https://github.com/Sanjana-dm)  [@luckynvdu](https://github.com/luckynvdu) [@surendrasinghs](https://github.com/surendrasinghs) [@rhwarrier](https://github.com/rhwarrier)
[@kameshbhr](https://github.com/kameshbhr) [@mmanimegalai](https://github.com/mmanimegalai) [@SanthoshVasabhaktula](https://github.com/SanthoshVasabhaktula) [@vrayulu](https://github.com/vrayulu) [@pritamps](https://github.com/pritamps) [@karn-nahata](https://github.com/karn-nahata) [@pkranga](https://github.com/pkranga) [@Prerna789](https://github.com/Prerna789) [@aayushkedia95](https://github.com/aayushkedia95) [@anandp504](https://github.com/anandp504) [@reshmi-nair](https://github.com/reshmi-nair) [@pallakartheekreddy](https://github.com/pallakartheekreddy) [@vinukumar-vs](https://github.com/vinukumar-vs) [@sharathkashyap](https://github.com/sharathkashyap) [aishwaryashikshalokam](https://github.com/aishwaryashikshalokam)

**Comment by Sanjana-dm** [comment]
> Dear all,
> 
> This is the document on the behaviour of resources on the platform based on their end and start date. Examples and use cases have been taken based on projects, observations, courses, programs and surveys. https://docs.google.com/document/d/1SU3EDHi_T2QZ2sodbH992B5db5XHgzbR-KYpvxFsDQs/edit?usp=sharing
> 
> [@Prateek-slokam](https://github.com/Prateek-slokam) [@aks30](https://github.com/aks30) [@kiranharidas187](https://github.com/kiranharidas187) [@vijiurs](https://github.com/vijiurs) [@Sanjana-dm](https://github.com/Sanjana-dm) [@luckynvdu](https://github.com/luckynvdu) [@surendrasinghs](https://github.com/surendrasinghs) [@rhwarrier](https://github.com/rhwarrier) [@kameshbhr](https://github.com/kameshbhr) [@mmanimegalai](https://github.com/mmanimegalai) [@SanthoshVasabhaktula](https://github.com/SanthoshVasabhaktula) [@vrayulu](https://github.com/vrayulu) [@pritamps](https://github.com/pritamps) [@karn-nahata](https://github.com/karn-nahata) [@pkranga](https://github.com/pkranga) [@Prerna789](https://github.com/Prerna789) [@aayushkedia95](https://github.com/aayushkedia95) [@anandp504](https://github.com/anandp504) [@reshmi-nair](https://github.com/reshmi-nair) [@pallakartheekreddy](https://github.com/pallakartheekreddy) [@vinukumar-vs](https://github.com/vinukumar-vs) [@sharathkashyap](https://github.com/sharathkashyap) [aishwaryashikshalokam](https://github.com/aishwaryashikshalokam)

Dear all, 

Requesting you all to leave your comments and suggestions, if any, on the document. 
We hope to take this solution forward soon.

**Reply by Krishnaj20** [reply]
@Sanjana-dm - I believe this RFC is about generalizing the program construct? I could not find details about the context or why these changes are needed. From the larger context what scope you are driving to contribute or you are planning to do everything in one shot.  

I'd recommend you to sync up with Varsha or go thru the below document where the discussions b/w Sunbird community and SL team are documented on how to approach this. 

https://docs.google.com/presentation/d/1-E76NhnxLZFGcLMap9aCzWmxlx2n_iimVmdsDtE40A8/edit?usp=sharing

**Comment by Sanjana-dm** [comment]
> Dear all,
> 
> This is the document on the behaviour of resources on the platform based on their end and start date. Examples and use cases have been taken based on projects, observations, courses, programs and surveys. https://docs.google.com/document/d/1SU3EDHi_T2QZ2sodbH992B5db5XHgzbR-KYpvxFsDQs/edit?usp=sharing
> 
> [@Prateek-slokam](https://github.com/Prateek-slokam) [@aks30](https://github.com/aks30) [@kiranharidas187](https://github.com/kiranharidas187) [@vijiurs](https://github.com/vijiurs) [@Sanjana-dm](https://github.com/Sanjana-dm) [@luckynvdu](https://github.com/luckynvdu) [@surendrasinghs](https://github.com/surendrasinghs) [@rhwarrier](https://github.com/rhwarrier) [@kameshbhr](https://github.com/kameshbhr) [@mmanimegalai](https://github.com/mmanimegalai) [@SanthoshVasabhaktula](https://github.com/SanthoshVasabhaktula) [@vrayulu](https://github.com/vrayulu) [@pritamps](https://github.com/pritamps) [@karn-nahata](https://github.com/karn-nahata) [@pkranga](https://github.com/pkranga) [@Prerna789](https://github.com/Prerna789) [@aayushkedia95](https://github.com/aayushkedia95) [@anandp504](https://github.com/anandp504) [@reshmi-nair](https://github.com/reshmi-nair) [@pallakartheekreddy](https://github.com/pallakartheekreddy) [@vinukumar-vs](https://github.com/vinukumar-vs) [@sharathkashyap](https://github.com/sharathkashyap) [aishwaryashikshalokam](https://github.com/aishwaryashikshalokam)

@aishwaryashikshalokam

**Reply by Krishnaj20** [reply]
@Sanjana-dm - is this a new version of the document?

**Comment by aishwaryashikshalokam** [comment]
Dear @Krishnaj20 

Apologies for late response. 

Programs are always created for a duration of time. As of now the behaviour of programs and assets within the program post it's end date is defined but is not there in development.

This document is created based on the discussions that happened in past related to Program and behavior of assets within the Program. It captures all the points that were discussed based on flows especially if there are start and end date given to it. 

We want to take the flows forward in development  and thus created this solution note. 

Based on your comments, I understand that you are proposing that if programs are expired we should not let users consume it. 
We are proposing that users can still complete the program so that there is no restriction for them to continue to learn and improve. But because they did not complete it in the given time frame, they will not be considered with the Program.

Let us know if more discussion is needed on this approach. We can schedule sometime on the same.

**Reply by Krishnaj20** [reply]
@aishwaryashikshalokam - Can we look at implementing a workflow similar to how course batch post expiry works? Once the course batch expires, users enrolled to the course can consume the content but their progress isn't updated hence they cannot complete the course after the batch has expired. Access to learning resources remains but the users cannot complete the course post batch expiry.

**Reply by aishwaryashikshalokam** [reply]
Hi @Krishnaj20 

Courses do not have explicit submit button. In observations and projects there is an explicit button that says submit. 

As per course behaviour, does it mean that users will be able to fill the the observation form and take up improvement projects but submit button for them will not get enabled if it is after the end date of the program/resource.

Should the behaviour be same if the resource end date is passed but program is still on going.

**Reply by aishwaryashikshalokam** [reply]
@Krishnaj20 any thoughts on the same?

**Reply by Krishnaj20** [reply]
@Prateek-slokam - Can you table this in next week's working group call? Will need wider team's inputs on this thread.

@vinukumar-vs @chitranshu-keshav @kiranharidas187 @SanthoshVasabhaktula @parthlawate

**Reply by Prateek-slokam** [reply]
@Krishnaj20 Sure. I will bring this up in the agenda for next Friday.
